### PR TITLE
[feature] Expand tilde to user `$HOME` when handling user input paths

### LINKED ISSUE #629: Support ~ as $HOME in all paths
Create a wrapper function to convert str->Path

### PR DESCRIPTION
**Pull request by pharmpy-dev-123**
Fixes #629.

### CODE REVIEW

File: src/pharmpy/config.py [hunk 1]
```diff
@@ -47,13 +47,14 @@ def read_configuration():
 
 
 class ConfigItem:
-    def __init__(self, default, description, cls=None):
+    def __init__(self, default, description, cls=None, transform=None):
```
Comment (pharmpy-dev-123) [inline src/pharmpy/config.py:50]: Collateral damage.

File: src/pharmpy/plugins/nonmem/__init__.py [hunk 2]
```diff
@@ -38,7 +39,10 @@ class NONMEMConfiguration(config.Configuration):
         list,
     )
     default_nonmem_path = config.ConfigItem(
-        pathlib.Path(''), 'Full path to the default NONMEM installation directory', cls=pathlib.Path
+        pathlib.Path(''),
+        'Full path to the default NONMEM installation directory',
+        cls=pathlib.Path,
+        transform=ux.normalize_user_given_path,
```
Comment (pharmpy-dev-123) [inline src/pharmpy/plugins/nonmem/__init__.py:45]: Not sure this belongs here.

File: src/pharmpy/workflows/execute.py [hunk 3]
```diff
@@ -102,6 +103,8 @@ def split_common_options(d):
     other_options = dict()
     for key, value in d.items():
         if key in execute_options:
+            if key == 'path':
+                value = normalize_user_given_path(value)
```
Comment (pharmpy-dev-123) [inline src/pharmpy/workflows/execute.py:107]: Not sure this belongs here either.

**Review by pharmpy-dev-123 — COMMENTED** [review]
I left some comments on the first draft.

File: src/pharmpy/plugins/nonmem/__init__.py (hunk 2 above)
Comment (rikardn) [reply src/pharmpy/plugins/nonmem/__init__.py:45]: I think you don't need the extra transform argument here. `cls` could be set to any callable returning an object of the class.

File: src/pharmpy/workflows/execute.py (hunk 3 above)
Comment (rikardn) [reply src/pharmpy/workflows/execute.py:107]: Makes sense to have here. Good catch

**Review by rikardn — CHANGES_REQUESTED** [review]

File: src/pharmpy/modeling/common.py [hunk 4]
```diff
@@ -11,6 +11,7 @@
 from pharmpy import Model, Parameters, RandomVariables, config
 from pharmpy.modeling import split_joint_distribution
 from pharmpy.statements import Assignment, CompartmentalSystem
+from pharmpy.ux import normalize_user_given_path
```
[the remark this answers is not in the document]
Comment (pharmpy-dev-123) [reply src/pharmpy/modeling/common.py:14]: Ok

**Comment by rikardn** [comment]
Thanks!

**Pull request merged**